src/main/java/com/cookingchef/dao/Postgres/PostgresCartDAO.java

```diff
@@ -0,0 +1,123 @@
+package com.cookingchef.dao.Postgres;
+
+import com.cookingchef.dao.CartDAO;
+import com.cookingchef.dbutils.ConnectionManager;
+import com.cookingchef.model.Cart;
+import com.cookingchef.model.CartDbFields;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class PostgresCartDAO implements CartDAO {
+    private static AtomicReference<PostgresCartDAO> instance = new AtomicReference<>();
+
+    private PostgresCartDAO() {
+    }
+
+    /**
+     * If the instance is null, create a new instance and return it.
+     * Otherwise, return the existing instance
+     *
+     * @return A PostgresCartDAO object.
+     */
+    public static PostgresCartDAO getPostgresCartDAO() {
+        instance.compareAndSet(null, new PostgresCartDAO());
+        return instance.get();
+    }
+
+    @Override
+    public Optional<Integer> addElementIntoCartInDb(Cart cart) throws SQLException {
+        var query = "INSERT INTO cart_user(ingredient_id, user_id, quantity,unit) VALUES(?, ?, ?, ?) RETURNING id";
+        var conn = ConnectionManager.getConnection();
+
+        try (var stmt = conn.prepareStatement(query)) {
+            stmt.setInt(1, cart.getIngredient_id());
+            stmt.setInt(2, cart.getUser_id());
+            stmt.setInt(3, cart.getQuantity());
+            stmt.setInt(4, cart.getUnit());
+
+            stmt.executeQuery();
+            var rs = stmt.getResultSet();
+            rs.next();
+            var newId = rs.getInt(CartDbFields.ID.value);
+            cart.setId(newId);
+            return Optional.of(newId);
+        }
+    }
+
+    @Override
+    public void updateCartInDb(Cart cart) throws SQLException {
+        var query = "UPDATE cart_user SET quantity = ? WHERE id = ?";
```

Comment:
You should add the unit too
```suggestion
        var query = "UPDATE cart_user SET quantity = ?, unit = ? WHERE ingredient_id = ? AND user_id = ?";
```